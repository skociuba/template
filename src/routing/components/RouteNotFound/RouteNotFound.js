import React from 'react';
import {Navigate} from 'react-router-dom';
import PropTypes from 'prop-types';

const RouteNotFound = ({children, path, redirectTo}) =>
  path === '*' ? <Navigate to={redirectTo} /> : children;

RouteNotFound.propTypes = {
  children: PropTypes.element.isRequired,
  path: PropTypes.string.isRequired,
  redirectTo: PropTypes.string.isRequired,
};

export default RouteNotFound;
