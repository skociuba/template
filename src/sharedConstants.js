import {routes} from './routesConstants';

const shared = {
  routes: routes,
  header: {
    menu: {
      primary: [
        {name: 'Main', to: '/main-page'},
        {name: 'FrontendPaginationSorting', to: '/frontend-pagination-sorting'},
        {name: 'StaticTable', to: '/static-table-page'},
        {name: 'BackendControl', to: '/backend-control-page'},
        {name: 'FrontendControl', to: '/frontend-control-page'},
      ],
    },
  },
  mapping: [{value: 'testData', label: 'MappedData'}],
  statuses: {
    B: {
      title: 'BUY',
    },
    S: {
      title: 'SELL',
    },
  },
  names: {
    '603be08a2987d5267f590d2f': {
      title: '603be08a2987d5267f590d2f',
      isSelected: false,
    },
    '603be21b2987d50c34590d38': {
      title: '603be21b2987d50c34590d38',
      isSelected: false,
    },
    all: {
      title: 'all',
      isSelected: false,
    },
  },
};

export {shared};
