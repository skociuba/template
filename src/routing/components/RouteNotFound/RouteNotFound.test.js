import renderWithRouter from 'utils/tests/renderWithRouter';
import {shared} from 'sharedConstants';

import RouteNotFound from './RouteNotFound';

const paths = {...shared.routes};

const defaultProps = {
  children: <div>Page</div>,
  path: paths.testPage.root,
  redirectTo: paths.testPage.root,
};

describe('<RouteNotFound/>', () => {
  it('render without crashing', () => {
    const {container} = renderWithRouter(<RouteNotFound {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
