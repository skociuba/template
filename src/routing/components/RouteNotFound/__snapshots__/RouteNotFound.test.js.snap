// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteNotFound/> render without crashing 1`] = `
<div>
  Page
</div>
`;
